### PR TITLE
Optimize UI update delay when opening TopMenuMobile

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Improve INP for opening `TopMenuMobile`
1. Modal and header renders
2. Menu content renders

Also replaced menu triggers with standard Button component, this simplifies styling

## Justify why they are needed

This leads to ~2x better INP on our top slow interaction for mobile users

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4256?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4256** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4256?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4255** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4255?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/blocks/HeaderBlock.tsx [hunk 1]
```diff
@@ -249,17 +248,6 @@ export type HeaderBlockProps = SbBaseBlockProps<{
 // - we render both desktop and mobile menu to allow CSS-only switch between them when window is resized
 // - desktop menu is memoized to avoid rerendering when mobile menu state changes
 export const HeaderBlock = ({ blok, ...headerProps }: HeaderBlockProps) => {
-  const pathname = usePathname()
```
Comment (alebedev) [inline apps/store/src/blocks/HeaderBlock.tsx:252]: This is state management improvement after previous PR, we no longer need to know TopMenuMobile.isOpen in header scope

File: apps/store/src/components/Header/TopMenuMobile/TopMenuMobile.tsx [hunk 2]
```diff
@@ -123,6 +132,7 @@ const TopMenuHeader = styled.div({
   justifyContent: 'space-between',
   height: MENU_BAR_HEIGHT_MOBILE,
   paddingInline: theme.space.md,
+  gap: theme.space.xs,
```
Comment (alebedev) [inline apps/store/src/components/Header/TopMenuMobile/TopMenuMobile.tsx:135]: Subtle bug in old version - we were loosing the gap when replacing main header with header within the modal

File: apps/store/src/components/Header/TopMenuMobile/TopMenuMobile.tsx [hunk 3]
```diff
@@ -37,63 +22,87 @@ const ButtonWrapper = styled.div({
...
 export const TopMenuMobile = (props: TopMenuMobileProps) => {
-  const { children, isOpen, onOpenChange, defaultValue } = props
+  const pathname = usePathname()
+  const [isOpen, setIsOpen] = useState(false)
+  const [isContentVisible, setIsContentVisible] = useState(false)
+  // Close on navigation
+  useEffect(() => {
+    setIsOpen(false)
+  }, [pathname])
+  const handleOpenChange = useCallback(
```
Comment (Youakeem) [inline apps/store/src/components/Header/TopMenuMobile/TopMenuMobile.tsx:37]: `DialogPrimitive.Root` is not memoized so I don't think we get value from `useCallback` here.

File: apps/store/src/components/Header/TopMenuMobile/TopMenuMobile.tsx [hunk 4]
```diff
@@ -37,63 +22,87 @@ const ButtonWrapper = styled.div({
...
+    (newValue: boolean) => startTransition(() => setIsOpen(newValue)),
+    [],
+  )
+  // Poor man's useDeferredValue that waits for underlying dialog's display animation
+  useEffect(() => {
```
Comment (Youakeem) [inline apps/store/src/components/Header/TopMenuMobile/TopMenuMobile.tsx:42]: Can we use `onAnimationEnd` in this case?

Comment (alebedev) [reply apps/store/src/components/Header/TopMenuMobile/TopMenuMobile.tsx:42]: We'd rather start rendering content ASAP after initial frame is rendered, there's no goal to sync with animation state

Comment (Youakeem) [reply apps/store/src/components/Header/TopMenuMobile/TopMenuMobile.tsx:42]: I see, maybe I misunderstood the goal. The comment says `waits for underlying dialog's display animation` which gave me the impression we are waiting for the animation to end.

**Review by guilhermespopolin — APPROVED** [review]
Nice refactoring 🙌🏻

**Review by Youakeem — APPROVED** [review]

File: apps/store/src/components/Header/TopMenuMobile/TopMenuMobile.tsx (hunk 4 above)
Comment (alebedev) [reply apps/store/src/components/Header/TopMenuMobile/TopMenuMobile.tsx:42]: Not that, just next UI frame.  I'll update the comment

**Comment by alebedev** [comment]
### Merge activity

* **May 13, 11:47 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4256) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**